src/abstract-interpretation/domains/positive-interval-domain.ts

```diff
@@ -49,13 +56,51 @@ export class PosIntervalDomain<Value extends PosIntervalLift = PosIntervalLift>
 		return PosIntervalDomain.bottom();
 	}
 
+	public join(...values: PosIntervalDomain[]): PosIntervalDomain;
+	public join(...values: PosIntervalLift[]): PosIntervalDomain;
```

Comment:
are these... necessary? You do not define a dependent type relation wrt. the return type so this is just the signature below